packages/react/polymorphic/src/forwardRefWithAs.ts

```diff
@@ -3,28 +3,34 @@ import * as React from 'react';
 /* -------------------------------------------------------------------------------------------------
  * Utility types
  * -----------------------------------------------------------------------------------------------*/
-type NeverKeys<T> = { [P in keyof T]: T[P] extends never ? P : never }[keyof T];
-type MergeProps<P1 = {}, P2 = {}> = Omit<P1, keyof P2> &
-  (NeverKeys<P2> extends never ? P2 : Omit<P2, NeverKeys<P2>>);
+type Merge<P1 = {}, P2 = {}> = Omit<P1, keyof P2> & P2;
```

Comment:
Is this the same as the one in our types utils? Perhaps it was duplicated on purpose to avoid the dependency?

Reply:
Yeah it's the same and that's exactly why I duplicated it. Utils package has a lot of stuff in it and seemed like an unnecessary overhead for this simple util.